olp-cpp-sdk-dataservice-write/src/IndexLayerClientImpl.h

```diff
@@ -50,6 +50,8 @@ class IndexLayerClientImpl
  public:
   IndexLayerClientImpl(client::HRN catalog, client::OlpClientSettings settings);
 
+  ~IndexLayerClientImpl();
```

Comment:
```suggestion
virtual ~IndexLayerClientImpl();
```
I suggest to mark it as virtual in case when we will add proper unit tests for those Implementation files and mock them. Same with other implementation files.

What do you think?

Reply:
Good idea, I'll add it.